### PR TITLE
correctly retrieve fortran bindings information for OpenMPI 1.7 and hi…

### PR DESCRIPTION
**Pull request by ggouaillardet**
…gher

in MTT::MPI::Install::Analyze::OMPI::Install

### CODE REVIEW

**Comment by ggouaillardet** [comment]
@jsquyres @rhc54 i found that when using the ```AlreadyInstalled``` module with OpenMPI master and the trivial test suite
fortran tests were simply not built because the fortran bindings were reported missing

**Comment by miked-mellanox** [comment]
:+1:

File: lib/MTT/MPI/Install/Analyze/OMPI.pm [hunk 1]
```diff
@@ -46,13 +48,33 @@ sub Install {
...
-    Debug("Have F77 bindings: $ret->{f77_bindings}\n"); 
-    $ret->{f90_bindings} = &{$func}($ret->{bindir}, $ret->{libdir}, "f90");
-    Debug("Have F90 bindings: $ret->{f90_bindings}\n"); 
+
+    # OMPI 1.7 (and higher) refer to "bindings:mpif.h".  Prior
+    # versions refer to "bindings:f77".
+    my $tmp;
+    $tmp = &{$func}($ret->{bindir}, $ret->{libdir}, "f77");
+    $tmp = &{$func}($ret->{bindir}, $ret->{libdir}, "mpif.h")
+        if (!$tmp);
```
Comment (jsquyres) [inline lib/MTT/MPI/Install/Analyze/OMPI.pm]: Should this be `if (!defined($tmp))`?

File: lib/MTT/MPI/Install/Analyze/OMPI.pm [hunk 2]
```diff
@@ -46,13 +48,33 @@ sub Install {
...
+    $ret->{mpifh_bindings} = $ret->{f77_bindings} = $tmp;
+    Debug("Have mpif.h bindings: $ret->{mpifh_bindings}\n"); 
+
+    # OMPI 1.7 (and higher) refer to "bindings:use_mpi".  Prior
+    # versions refer to "bindings:f90".
+    $tmp = &{$func}($ret->{bindir}, $ret->{libdir}, "f90");
+    $tmp = &{$func}($ret->{bindir}, $ret->{libdir}, "use_mpi")
+        if (!$tmp);
```
Comment (jsquyres) [inline lib/MTT/MPI/Install/Analyze/OMPI.pm]: Similar to above: `if (!defined($tmp))`

**Comment by ggouaillardet** [comment]
This is a copy/paste from https://github.com/open-mpi/mtt/blob/master/lib/MTT/MPI/Install/OMPI.pm

```perl
sub find_bindings {
    my ($bindir, $libdir, $lang) = @_;

    my $file = _run_ompi_info($bindir, $libdir, "^bindings:$lang:");
    return ($file->[0] =~ /^bindings:${lang}:(")?yes/) ? "1" : "0";
}
```
find_bindings returns "0" or "1" so I think the code is currently correct

**Comment by jsquyres** [comment]
@ggouaillardet Fair enough. 

FWIW: I prefer a defensive programming style -- unexpectedly undefined perl vars can lead to difficult-to-find bugs later.  As such, I like to guard for them whenever possible.

**Comment by ggouaillardet** [comment]
so the correct statement should be
```perl
... if (!defined($tmp) || !$tmp);
```
or maybe in a more compact and perlish form
```perl
... unless($tmp);
```

I do not mind updating the PR to avoid bugs in the future

**Comment by jsquyres** [comment]
Yes, I think you're right: `if (!defined($tmp) || !$tmp)`.

Cool; thanks.

**Comment by jsquyres** [comment]
Wait, I missed your `unless` form... does `unless` also check for `!defined`?  I never keep that stuff straight.  As you can probably tell, I write perl very much like I write C.  :smile:

**Comment by ggouaillardet** [comment]
@jsquyres done
finally, i replaced ```if (!$tmp)``` with ```unless(defined($tmp) && $tmp)```
omitting the ```defined()``` test will likely generate a warning if the variable is not defined

**Comment by jsquyres** [comment]
:+1:

**Pull request merged**